// @mirohq/miro-api/dist/model/appCardDataChanges.js
"use strict";
/**
 * Miro Developer Platform
 * <img src=\"https://content.pstmn.io/47449ea6-0ef7-4af2-bac1-e58a70e61c58/aW1hZ2UucG5n\" width=\"1685\" height=\"593\">  ### Miro Developer Platform concepts  - New to the Miro Developer Platform? Interested in learning more about platform concepts?? [Read our introduction page](https://beta.developers.miro.com/docs/introduction) and familiarize yourself with the Miro Developer Platform capabilities in a few minutes.   ### Getting started with the Miro REST API  - [Quickstart (video):](https://beta.developers.miro.com/docs/try-out-the-rest-api-in-less-than-3-minutes) try the REST API in less than 3 minutes. - [Quickstart (article):](https://beta.developers.miro.com/docs/build-your-first-hello-world-app-1) get started and try the REST API in less than 3 minutes.   ### Miro REST API tutorials  Check out our how-to articles with step-by-step instructions and code examples so you can:  - [Get started with OAuth 2.0 and Miro](https://beta.developers.miro.com/docs/getting-started-with-oauth)   ### Miro App Examples  Clone our [Miro App Examples repository](https://github.com/miroapp/app-examples) to get inspiration, customize, and explore apps built on top of Miro\'s Developer Platform 2.0.
 *
 * The version of the OpenAPI document: v2.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppCardDataChanges = void 0;
/**
 * @internal
 * Contains app card item data, such as the title, description, or fields.
 */
class AppCardDataChanges {
    constructor() {
        /**
         * Status indicating whether an app card is connected and in sync with the source. When the source for the app card is deleted, the status returns `disabled`.
         */
        this['status'] = AppCardDataChanges.StatusEnum.Disconnected;
        /**
         * A short text header to identify the app card.
         */
        this['title'] = 'sample app card item';
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return AppCardDataChanges.attributeTypeMap;
    }
}
exports.AppCardDataChanges = AppCardDataChanges;
/** @ignore */
AppCardDataChanges.discriminator = undefined;
/** @ignore */
AppCardDataChanges.attributeTypeMap = [
    {
        name: 'description',
        baseName: 'description',
        type: 'string',
    },
    {
        name: 'fields',
        baseName: 'fields',
        type: 'Array<CustomField>',
    },
    {
        name: 'status',
        baseName: 'status',
        type: 'AppCardDataChanges.StatusEnum',
    },
    {
        name: 'title',
        baseName: 'title',
        type: 'string',
    },
];
(function (AppCardDataChanges) {
    AppCardDataChanges.StatusEnum = {
        Disconnected: 'disconnected',
        Connected: 'connected',
        Disabled: 'disabled',
    };
})(AppCardDataChanges = exports.AppCardDataChanges || (exports.AppCardDataChanges = {}));
