// @mirohq/miro-api/dist/model/connectorWithLinks.js
"use strict";
/**
 * Miro Developer Platform
 * <img src=\"https://content.pstmn.io/47449ea6-0ef7-4af2-bac1-e58a70e61c58/aW1hZ2UucG5n\" width=\"1685\" height=\"593\">  ### Miro Developer Platform concepts  - New to the Miro Developer Platform? Interested in learning more about platform concepts?? [Read our introduction page](https://beta.developers.miro.com/docs/introduction) and familiarize yourself with the Miro Developer Platform capabilities in a few minutes.   ### Getting started with the Miro REST API  - [Quickstart (video):](https://beta.developers.miro.com/docs/try-out-the-rest-api-in-less-than-3-minutes) try the REST API in less than 3 minutes. - [Quickstart (article):](https://beta.developers.miro.com/docs/build-your-first-hello-world-app-1) get started and try the REST API in less than 3 minutes.   ### Miro REST API tutorials  Check out our how-to articles with step-by-step instructions and code examples so you can:  - [Get started with OAuth 2.0 and Miro](https://beta.developers.miro.com/docs/getting-started-with-oauth)   ### Miro App Examples  Clone our [Miro App Examples repository](https://github.com/miroapp/app-examples) to get inspiration, customize, and explore apps built on top of Miro\'s Developer Platform 2.0.
 *
 * The version of the OpenAPI document: v2.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectorWithLinks = void 0;
/**
 * @internal
 * Contains the result data.
 */
class ConnectorWithLinks {
    constructor() {
        /**
         * The path type of the connector line, defines curvature. Default: curved.
         */
        this['shape'] = ConnectorWithLinks.ShapeEnum.Curved;
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return ConnectorWithLinks.attributeTypeMap;
    }
}
exports.ConnectorWithLinks = ConnectorWithLinks;
/** @ignore */
ConnectorWithLinks.discriminator = undefined;
/** @ignore */
ConnectorWithLinks.attributeTypeMap = [
    {
        name: 'captions',
        baseName: 'captions',
        type: 'Array<Caption>',
    },
    {
        name: 'createdAt',
        baseName: 'createdAt',
        type: 'Date',
    },
    {
        name: 'createdBy',
        baseName: 'createdBy',
        type: 'CreatedBy',
    },
    {
        name: 'endItem',
        baseName: 'endItem',
        type: 'ItemConnectionWithLinks',
    },
    {
        name: 'id',
        baseName: 'id',
        type: 'string',
    },
    {
        name: 'isSupported',
        baseName: 'isSupported',
        type: 'boolean',
    },
    {
        name: 'links',
        baseName: 'links',
        type: 'SelfLink',
    },
    {
        name: 'modifiedAt',
        baseName: 'modifiedAt',
        type: 'Date',
    },
    {
        name: 'modifiedBy',
        baseName: 'modifiedBy',
        type: 'ModifiedBy',
    },
    {
        name: 'shape',
        baseName: 'shape',
        type: 'ConnectorWithLinks.ShapeEnum',
    },
    {
        name: 'startItem',
        baseName: 'startItem',
        type: 'ItemConnectionWithLinks',
    },
    {
        name: 'style',
        baseName: 'style',
        type: 'ConnectorStyle',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
(function (ConnectorWithLinks) {
    ConnectorWithLinks.ShapeEnum = {
        Straight: 'straight',
        Elbowed: 'elbowed',
        Curved: 'curved',
    };
})(ConnectorWithLinks = exports.ConnectorWithLinks || (exports.ConnectorWithLinks = {}));
