// @mirohq/miro-api/dist/model/teamSharingPolicySettings.js
"use strict";
/**
 * Miro Developer Platform
 * <img src=\"https://content.pstmn.io/47449ea6-0ef7-4af2-bac1-e58a70e61c58/aW1hZ2UucG5n\" width=\"1685\" height=\"593\">  ### Miro Developer Platform concepts  - New to the Miro Developer Platform? Interested in learning more about platform concepts?? [Read our introduction page](https://beta.developers.miro.com/docs/introduction) and familiarize yourself with the Miro Developer Platform capabilities in a few minutes.   ### Getting started with the Miro REST API  - [Quickstart (video):](https://beta.developers.miro.com/docs/try-out-the-rest-api-in-less-than-3-minutes) try the REST API in less than 3 minutes. - [Quickstart (article):](https://beta.developers.miro.com/docs/build-your-first-hello-world-app-1) get started and try the REST API in less than 3 minutes.   ### Miro REST API tutorials  Check out our how-to articles with step-by-step instructions and code examples so you can:  - [Get started with OAuth 2.0 and Miro](https://beta.developers.miro.com/docs/getting-started-with-oauth)   ### Miro App Examples  Clone our [Miro App Examples repository](https://github.com/miroapp/app-examples) to get inspiration, customize, and explore apps built on top of Miro\'s Developer Platform 2.0.
 *
 * The version of the OpenAPI document: v2.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TeamSharingPolicySettings = void 0;
/**
 * @internal
 * Team sharing policy settings
 */
class TeamSharingPolicySettings {
    /** @ignore */
    static getAttributeTypeMap() {
        return TeamSharingPolicySettings.attributeTypeMap;
    }
}
exports.TeamSharingPolicySettings = TeamSharingPolicySettings;
/** @ignore */
TeamSharingPolicySettings.discriminator = undefined;
/** @ignore */
TeamSharingPolicySettings.attributeTypeMap = [
    {
        name: 'allowListedDomains',
        baseName: 'allowListedDomains',
        type: 'Array<string>',
    },
    {
        name: 'createAssetAccessLevel',
        baseName: 'createAssetAccessLevel',
        type: 'TeamSharingPolicySettings.CreateAssetAccessLevelEnum',
    },
    {
        name: 'defaultBoardAccess',
        baseName: 'defaultBoardAccess',
        type: 'TeamSharingPolicySettings.DefaultBoardAccessEnum',
    },
    {
        name: 'defaultBoardSharingAccess',
        baseName: 'defaultBoardSharingAccess',
        type: 'TeamSharingPolicySettings.DefaultBoardSharingAccessEnum',
    },
    {
        name: 'defaultOrganizationAccess',
        baseName: 'defaultOrganizationAccess',
        type: 'TeamSharingPolicySettings.DefaultOrganizationAccessEnum',
    },
    {
        name: 'defaultProjectAccess',
        baseName: 'defaultProjectAccess',
        type: 'TeamSharingPolicySettings.DefaultProjectAccessEnum',
    },
    {
        name: 'moveBoardToAccount',
        baseName: 'moveBoardToAccount',
        type: 'TeamSharingPolicySettings.MoveBoardToAccountEnum',
    },
    {
        name: 'restrictAllowedDomains',
        baseName: 'restrictAllowedDomains',
        type: 'TeamSharingPolicySettings.RestrictAllowedDomainsEnum',
    },
    {
        name: 'sharingOnAccount',
        baseName: 'sharingOnAccount',
        type: 'TeamSharingPolicySettings.SharingOnAccountEnum',
    },
    {
        name: 'sharingOnOrganization',
        baseName: 'sharingOnOrganization',
        type: 'TeamSharingPolicySettings.SharingOnOrganizationEnum',
    },
    {
        name: 'sharingViaPublicLink',
        baseName: 'sharingViaPublicLink',
        type: 'TeamSharingPolicySettings.SharingViaPublicLinkEnum',
    },
];
(function (TeamSharingPolicySettings) {
    TeamSharingPolicySettings.CreateAssetAccessLevelEnum = {
        CompanyAdmins: 'company_admins',
        Admins: 'admins',
        AllMembers: 'all_members',
    };
    TeamSharingPolicySettings.DefaultBoardAccessEnum = {
        Private: 'private',
        View: 'view',
        Comment: 'comment',
        Edit: 'edit',
    };
    TeamSharingPolicySettings.DefaultBoardSharingAccessEnum = {
        TeamMembersWithEditingRights: 'team_members_with_editing_rights',
        OwnerAndCoowners: 'owner_and_coowners',
    };
    TeamSharingPolicySettings.DefaultOrganizationAccessEnum = {
        Private: 'private',
        View: 'view',
        Comment: 'comment',
        Edit: 'edit',
    };
    TeamSharingPolicySettings.DefaultProjectAccessEnum = {
        Private: 'private',
        View: 'view',
    };
    TeamSharingPolicySettings.MoveBoardToAccountEnum = {
        Allowed: 'allowed',
        NotAllowed: 'not_allowed',
    };
    TeamSharingPolicySettings.RestrictAllowedDomainsEnum = {
        Enabled: 'enabled',
        EnabledWithExternalUserAccess: 'enabled_with_external_user_access',
        Disabled: 'disabled',
    };
    TeamSharingPolicySettings.SharingOnAccountEnum = {
        Allowed: 'allowed',
        NotAllowed: 'not_allowed',
    };
    TeamSharingPolicySettings.SharingOnOrganizationEnum = {
        Allowed: 'allowed',
        AllowedWithEditing: 'allowed_with_editing',
        NotAllowed: 'not_allowed',
    };
    TeamSharingPolicySettings.SharingViaPublicLinkEnum = {
        Allowed: 'allowed',
        AllowedWithEditing: 'allowed_with_editing',
        NotAllowed: 'not_allowed',
    };
})(TeamSharingPolicySettings = exports.TeamSharingPolicySettings || (exports.TeamSharingPolicySettings = {}));
